lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -124,11 +112,13 @@ public function getExtension()
      */
     protected function getFilename($id)
     {
+        $hash = hash('sha256', $id);
```

Comment:
I realize that the likeliness of a collision is pretty much zero, but since there is a chance, there must be a mechanism to check integrity.

 > No idea what you are talking about.

For example, https://github.com/doctrine/cache/blob/1c3483678ab767c3e21d3a96d6dd1006b327d425/lib/Doctrine/Common/Cache/FilesystemCache.php#L106 would have to save the key as part of the data, in order to check the key after retrieval.